dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+    }
+
+    if (element.name.local == 'g') {
+      double opacity = transform.opacity;
+      if (_hasAttr(element, 'opacity'))
+        opacity *= double.parse(_extractAttr(element, 'opacity'));
+
+      Matrix3 transformMatrix = transform.transformMatrix;
+      if (_hasAttr(element, 'transform'))
+        transformMatrix = transformMatrix.multiplied(
+            _parseSvgTransform(_extractAttr(element, 'transform')));
+
+      final _Transform subtreeTransform = new _Transform(
+        transformMatrix: transformMatrix,
+        opacity: opacity
+      );
+      paths.addAll(_interpretSvgGroup(element.children, subtreeTransform));
+    }
+  }
+  return paths;
+}
+
+// Given a points list in the form e.g: "25.0, 1.0 12.0, 12.0 23.0, 9.0" matches
+// the coordinated of the first point and the rest of the string, for the
+// example above:
+// group 1 will match "25.0"
+// group 2 will match "1.0"
+// group 3 will match "12.0, 12.0 23.0, 9.0"
+final RegExp _pointMatcher = new RegExp(r' *([\-\.0-9]+) *, *([\-\.0-9]+)(.*)');
+
+/// Parse a string with a list of points, e.g:
+/// '25.0, 1.0 12.0, 12.0 23.0, 9.0' will be parsed to:
+/// [Point(25.0, 1.0), Point(12.0, 12.0), Point(23.0, 9.0)].
+List<Point<double>> parsePoints(String points) {
+  String unParsed = points;
+  final List<Point<double>> result = <Point<double>>[];
+  while(unParsed.isNotEmpty && _pointMatcher.hasMatch(unParsed)) {
+    final Match m = _pointMatcher.firstMatch(unParsed);
+    result.add(new Point<double>(
+        double.parse(m.group(1)),
+        double.parse(m.group(2))
+    ));
+    unParsed = m.group(3);
+  }
+  return result;
+}
+
+/// Data for a single animation frame.
+class FrameData {
+  const FrameData(this.size, this.paths);
+
+  final Point<double> size;
+  final List<SvgPath> paths;
+
+  @override
+  bool operator ==(Object other){
+    if (runtimeType != other.runtimeType)
+      return false;
+    final FrameData typedOther = other;
+    return size == typedOther.size
+        && const ListEquality<SvgPath>().equals(paths, typedOther.paths);
+  }
+
+  @override
+  int get hashCode =>
+      size.hashCode ^
+      paths.hashCode;
+
+  @override
+  String toString() {
+    return 'FrameData(size: $size, paths: $paths)';
+  }
+}
+
+/// Represents an SVG path element.
+class SvgPath {
+  const SvgPath(this.id, this.commands, {this.opacity = 1.0});
+
+  final String id;
+  final List<SvgPathCommand> commands;
+  final double opacity;
+
+  static final RegExp _pathCommandMatcher = new RegExp(r'([MmLlQqCcAZz]) *([\-\.0-9 ,]*)');
+
+  static SvgPath fromElement(XmlElement pathElement) {
+    assert(pathElement.name.local == 'path');
+    final String id = _extractAttr(pathElement, 'id');
+    final String dAttr = _extractAttr(pathElement, 'd');
+    final List<SvgPathCommand> commands = <SvgPathCommand> [];
+    final SvgPathCommandBuilder commandsBuilder = new SvgPathCommandBuilder();
+    for (Match match in _pathCommandMatcher.allMatches(dAttr)) {
```

Comment:
as far as i can tell this is just going to skip over unknown path commands, which seems dangerous. I recommend a parser that just walks over the string, finding commands and extracting the points from those, failing on unknown commands.

Reply:
It is going to fail later when generating the code (see line 163)